apps/root/src/state/token-lists/actions.ts

```diff
@@ -38,11 +38,16 @@ export const startFetchingTokenLists = createAppAsyncThunk(
 export const fetchTokenDetails = createAppAsyncThunk<Token, { tokenAddress: string; chainId: number }>(
   'tokenLists/fetchTokenDetails',
   async ({ tokenAddress, chainId }, { dispatch, getState, extra: { web3Service } }) => {
-    const id = `${chainId}-${tokenAddress}` as TokenListId;
-    const tokensLists = getState().tokenLists.byUrl;
+    const id = `${chainId}-${tokenAddress.toLowerCase()}` as TokenListId;
+    const { byUrl: tokensLists, customTokens } = getState().tokenLists;
 
     const tokenList = parseTokenList({
-      tokensLists,
+      tokensLists: {
+        ...tokensLists,
+        'custom-tokens': customTokens,
+      },
+      chainId,
+      curateList: true,
```

Comment:
I'm not sure we want to curate this here, since we are asking for a specific token we should probably return it either if it was curated or not